### PR TITLE
Add affinity group support in exoscale driver

### PR DESCRIPTION
**Pull request by cab105**
Based on the suggestion from #3547 since I don't have write access to @runseb's repo, I'm creating new PR.

### CODE REVIEW

**Comment by nathanleclaire** [comment]
@vincentbernat Does it LGTY?

**Comment by vincentbernat** [comment]
Codewise, it's OK. However, by putting everyone in the docker-machine anti-affinity group by default, it means that a user spawning more than 10 docker machines will get an error about reaching the limit of the anti-affinity group size. I think that's a good default to have a default anti-affinity group, we just need to add a tidbit in the documentation to mention the case and the solution would be to override the default anti-affinity group. Something like:

> By default, machines are spawned in the `docker-machine` anti-affinity group. There is a limit in the number of machines that can be part of the same anti-affinity group. If you reach the limit, just specify another anti-affinity group with `--exoscale-affinity-group=docker-machine2`.

File: docs/drivers/exoscale.md [hunk 1]
```diff
@@ -32,6 +32,8 @@ Options:
 
 If a custom security group is provided, you need to ensure that you allow TCP ports 22 and 2376 in an ingress rule. Moreover, if you want to use Swarm, also add TCP port 3376.
 
+There is a limit to the number of anti-affinity groups a machine can belong to.  You can work around it by specifying a second anti-affinity group using `--exoscale-affinity-group=docker-machineX`
```
Comment (vincentbernat) [inline docs/drivers/exoscale.md]: It's the reverse. There is a limit to the number of machines an anti-affinity group can contain.

**Comment by vincentbernat** [comment]
IMO, the wording is still incorrect. Sorry for the nitpick. You say "There is a limit to the number of machines that an anti-affinity group can belong to", you should say "There is a limit to the number of machines that an anti-affinity group can have". You'll also need to squash the commits.

**Comment by cab105** [comment]
Its okay.  Wouldn't want to make a substandard commit.

**Comment by nathanleclaire** [comment]
Is this good to go @vincentbernat @cab105 ?

**Comment by vincentbernat** [comment]
LGTM, sorry for the delay!

**Comment by nathanleclaire** [comment]
cool -- LGTM

**Pull request merged**